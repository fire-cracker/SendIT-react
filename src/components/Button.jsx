import React from 'react';
import PropTypes from 'prop-types';

const Button = ({
  submit,
  onClick,
  disabled,
  value,
  className
}) => (
  <button
    type={submit}
    className={className}
    onClick={onClick}
    disabled={disabled}
  >
    {value}
  </button>
);

Button.propTypes = {
  submit: PropTypes.string.isRequired,
  onClick: PropTypes.func,
  disabled: PropTypes.bool,
  value: PropTypes.string.isRequired,
  className: PropTypes.string.isRequired,
};


export default Button;
