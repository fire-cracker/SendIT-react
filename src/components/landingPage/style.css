html {
  height: 100%;
}

* {
  box-sizing: border-box;
}

/*Styling for the body background*/
body {
  margin: 0px;
  width: 100%;
  height: 100%;
  background-repeat: no-repeat;
  background-size: 100% 100%;
  background-image: linear-gradient(rgba(0, 0, 0, .5), rgba(0, 0, 0, .5)), url('../../../public/img/background4.jpg');

}

h3 {
  font-family: sofia;
  font-size: 22px;
  margin: 0px;
}


.whole {
  height: 100vh;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  flex-flow: row wrap;
}

.heroWhite {
  display: block;
  color: white;
  padding-left: 2rem;
  padding-right: 2rem;
  width: 50%;
  font-family: 'Rubik', sans-serif;
  font-size: 1.5rem;
}

.containerDemo {
  width: 60%;
}

.password-error-message {
  color: #FBE15D;
}

.red-bg {
  border: 1px solid red;
  background: #fed;
}
/* Set a style for all buttons */
.loginButton,
.signupButton {
  background-color: rgba(0, 0, 0, 0.705);
  color: white;
  padding: 14px 20px;
  margin: 8px 0;
  border: none;
  cursor: pointer;
  width: 100%;
  border-radius: 1rem;
  outline:none;
}

.loginButton,
.signupButton:hover {
  opacity: 0.8;
}

/*Styling for the Footer section*/
#bottom {
  position: fixed;
  left: 0;
  bottom: 0;
  width: 100%;
}

footer {
  background-color: rgb(76, 70, 118);
  text-align: center;
  display: flex;
  flex-direction: column;
}

#footerContent {
  padding-left: 80px;
  padding-right: 80px;
}

#terms {
  float: left;
  color: black;
  font-size: 12px;
  text-align: center;
}

#socialIcons {
  float: right;
  font-size: 0px;
}

#socialIcons a img {
  padding-left: 5px;
  padding-right: 5px;
  margin-top: 7px;
}

#license {
  background-color: rgb(76, 70, 118);
  width: 100%;
  font-family: Georgia, 'Times New Roman', Times, serif;
  font-size: 15px;
  text-align: center;
  color: white;
  align-content: center;
  bottom: 0;
}


/* Styling the login/Signin container background */
form div {
  padding: 0px;
}

#register,
#login {
  margin: auto;
  overflow: auto;
  top: 0px;
  width: 100%;
  height: 60%;
  padding: 1% 1% 1% 1%;
  border: 2px solid grey;
  border-radius: 10px;
}

input {
  outline: none;
}

.to_register {
  color: whitesmoke;
}

#loginBtn,
#submitBtn {
  border-radius: 10px;
  text-align: center;
}

/* Styling for the username and password input fields */
#username,
.password,
#confirmPassword,
.email {
  width: 100%;
  padding: 5px 20px;
  margin: 8px 0;
  height: 40px;
  display: inline-block;
  border: 1px solid #ccc;
  box-sizing: border-box;
  border-radius: 5px;
}

/* #confirmPassword2{
    margin: 90px 0px 0px 0px;
  } */

/*Styling for the Password Validation Message which pops out when writting password*/
#message {
  display: none;
  background: #f1f1f1;
  color: #000;
  opacity: 0.7;
  float: left;
  left: 54%;
  width: 17%;
  margin: 11% 40% 0% 5%;
  overflow: auto;
  position: fixed;
  top: 0px;
  padding: 1% 1% 1% 1%;
}

#message h3 {
  text-align: left;
  font-size: 18px;
}

span.notMember {
  font-family: Arial, Helvetica, sans-serif;
  font-size: 20px;
}

.container {
  text-align: center;
  padding: 1%;
  color: white;
}

.wrapper {
  width:100%;
  display: flex;
  flex-direction: row;

}

/* Styling to add a green text color and a tick when the requirements are met */
.valid {
  color: green;
}

.valid:before {
  position: relative;
  left: -15px;
  content: "✔";
}

.password {
  content: "✔";
  color: green;
}

/* Styling to add a green text color and an "✖"  when the requirements are not met */
.invalid {
  color: red;
}

.invalid:before {
  position: relative;
  left: -15px;
  content: "✖";
}

#comparePassword {
  padding-top: 5px;
}


/* Center the position of the login and signup header */
.img_login {
  text-align: center;
  position: relative;
  padding: 0px;
  margin: 0px 0px;
  height: 30px;
}

h1 {
  margin: 0px;
  color: #f1f1f1;
}

div {
  color: #f1f1f1;
}

.loginParticulars {
  padding: 10px;
  text-align: left;
}

/* For desktop: */
@media only screen and (min-width: 768px) {

  #register,
  #login {
    width: 50%;
  }

  .whole {
    flex-direction: row;
  }
  .containerDemo {
    width: 50%;
  }
}


/* For tablets:
    @media only screen and (min-width: 600px) {
      #register, 
      #login{
            
        margin: 5% 20% 0% 5%;
        width: 60%;
        height: 100%;
         }
    } */