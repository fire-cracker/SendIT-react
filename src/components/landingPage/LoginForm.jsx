import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { toast } from 'react-toastify';
import { isAlphanumeric, isEmail } from 'validator';

import Button from '../Button';

class Login extends Component {
  state = {
    email: '',
    password: '',
    passwordErrorMessage: '',
    emailErrorMessage: '',
  };

  handleOnChangeEmail = (e) => {
    this.setState({
      email: e.target.value,
      emailErrorMessage: ''
    });
  };

  handleOnChangePassword = (e) => {
    this.setState({
      password: e.target.value,
      passwordErrorMessage: '',
    });
  }

  onPasswordBlur = () => {
    const { password } = this.state;

    if (!password.trim()) {
      return this.setState({
        passwordErrorMessage: 'Please enter your password.'
      });
    }

    if (!isAlphanumeric(password.trim())) {
      return this.setState({
        passwordErrorMessage: 'Use numbers and letters for password.'
      });
    }

    if (password.length < 8 || password.length > 15) {
      return this.setState({
        passwordErrorMessage: 'Password must between 8 and 15 characters.'
      });
    }
  };

  onEmailBlur = () => {
    const { email } = this.state;

    if (!email.trim()) {
      return this.setState({
        emailErrorMessage: 'Please enter your email.'
      });
    }

    if (!isEmail(email.trim())) {
      return this.setState({
        emailErrorMessage: 'email is invalid.'
      });
    }
  };

  hasFormErrors = () => {
    const {
      passwordErrorMessage,
      emailErrorMessage
    } = this.state;

    if (passwordErrorMessage || emailErrorMessage) {
      return true;
    }
    return false;
  }

  hasEmptyRequiredFields = () => {
    const {
      email,
      password
    } = this.state;

    if (!email || !password) {
      return true;
    }
    return false;
  }

  handleOnClickLogin = (e) => {
    e.preventDefault();
    const { history } = this.props;
    const { email, password } = this.state;
    this.props.login({ email, password })
      .then(() => {
        history.push('/createOrder');
      })
      .catch((error) => {
        const { response, response: { status, data } } = error;
        if (response && status === 400) {
          return toast.error(
            data.Error_Log.userEmail || data.Error_Log.userPassword
          );
        }
        if (response && status === 401) {
          return toast.error('Incorrect login credentials');
        }
        if (response && status === 404) {
          return toast.error('Incorrect login credentials');
        }
        toast.error('Unknown error');
      });
    this.setState({
      email: '',
      password: ''
    });
  };


  render() {
    const {
      email, password, passwordErrorMessage, emailErrorMessage
    } = this.state;
    const {
      showPassword,
      tooglePassword,
      toggleLoginSignup
    } = this.props;

    const buttonDisabled = this.hasFormErrors() || this.hasEmptyRequiredFields();

    return (
      <div id="login" className="animate form">
        <form
          className="login zoom"
          id="LoginForm"
          autoComplete="on"
        >
          <div className="img_login">
            <h1>Log in</h1>
          </div>

          <div className="loginParticulars">
            <div><b>Email Address</b></div>
            <small className="password-error-message">
            &nbsp;
              {emailErrorMessage}
            </small>
            <input
              type="text"
              id="loginEmail"
              className={`email ${emailErrorMessage && 'red-bg'}`}
              placeholder="Enter Email Address"
              value={email}
              name="email"
              onBlur={this.onEmailBlur}
              onChange={this.handleOnChangeEmail}
            />
            <div><b>Password</b></div>
            <small className="password-error-message">
            &nbsp;
              {passwordErrorMessage}
            </small>
            <input
              type={showPassword ? 'text' : 'password'}
              className={`password ${passwordErrorMessage && 'red-bg'}`}
              id="loginPassword"
              placeholder="Enter Password"
              value={password}
              name="password"
              onBlur={this.onPasswordBlur}
              onChange={this.handleOnChangePassword}
            />
            <div>
              <input
                type="checkbox"
                checked={showPassword}
                className="showPassword"
                id="loginShowPassword"
                name="remember"
                onChange={tooglePassword}
              />
              Show Password
            </div>
            <Button
              className="loginButton"
              id="loginBtn"
              type="submit"
              disabled={buttonDisabled}
              onClick={this.handleOnClickLogin}
              value="Login"
            />

            <div className="container">
              <div className="change_link">
                Not a member yet ?
                <a
                  href="#toregister"
                  className="to_register"
                  id="to_register"
                  onClick={toggleLoginSignup}
                >
                  Join us

                </a>
              </div>
            </div>
          </div>
        </form>
      </div>
    );
  }
}


Login.propTypes = {
  toggleLoginSignup: PropTypes.func,
  tooglePassword: PropTypes.func,
  login: PropTypes.func,
  showPassword: PropTypes.bool,
  history: PropTypes.object
};

export default Login;
