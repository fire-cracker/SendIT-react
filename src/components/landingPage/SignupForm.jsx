import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { toast } from 'react-toastify';
import { isAlphanumeric, isEmail } from 'validator';

import Button from '../Button';

export default class Signup extends Component {
  state = {
    username: '',
    email: '',
    password: '',
    passwordErrorMessage: '',
    emailErrorMessage: '',
    usernameErrorMessage: ''

  };

  handleOnChangeUsername = (e) => {
    this.setState({
      username: e.target.value,
      usernameErrorMessage: ''
    });
  };

  handleOnChangeEmail = (e) => {
    this.setState({
      email: e.target.value,
      emailErrorMessage: ''
    });
  };

  handleOnChangePassword = (e) => {
    this.setState({
      password: e.target.value,
      passwordErrorMessage: ''
    });
  };

  onUsernameBlur = () => {
    const { username } = this.state;

    if (!username) {
      return this.setState({
        usernameErrorMessage: 'Please enter your username.'
      });
    }

    if (!isAlphanumeric(username)) {
      return this.setState({
        usernameErrorMessage: 'Use numbers and letters for password.'
      });
    }

    if (username.length < 8 || username.length > 15) {
      return this.setState({
        usernameErrorMessage: 'Password must between 8 and 15 characters.'
      });
    }
  };

  onPasswordBlur = () => {
    const { password } = this.state;

    if (!password) {
      return this.setState({
        passwordErrorMessage: 'Please enter your password.'
      });
    }

    if (!isAlphanumeric(password)) {
      return this.setState({
        passwordErrorMessage: 'Use numbers and letters for password.'
      });
    }

    if (password.length < 8 || password.length > 15) {
      return this.setState({
        passwordErrorMessage: 'Password must between 8 and 15 characters.'
      });
    }
  };

  onEmailBlur = () => {
    const { email } = this.state;

    if (!email) {
      return this.setState({
        emailErrorMessage: 'Please enter your email.'
      });
    }

    if (!isEmail(email)) {
      return this.setState({
        emailErrorMessage: 'email is invalid.'
      });
    }
  };

  hasFormErrors = () => {
    const {
      passwordErrorMessage,
      emailErrorMessage,
      usernameErrorMessage
    } = this.state;

    if (passwordErrorMessage || emailErrorMessage || usernameErrorMessage) {
      return true;
    }
    return false;
  }

  hasEmptyRequiredFields = () => {
    const {
      username,
      email,
      password
    } = this.state;

    if (!username || !email || !password) {
      return true;
    }
    return false;
  }

  handleOnClickSignup = (e) => {
    e.preventDefault();
    const { history } = this.props;
    const {
      username,
      email,
      password
    } = this.state;

    this.props.signup({ username, email, password })
      .then(() => {
        history.push('/createOrder');
      })
      .catch((error) => {
        const { response, response: { status, data } } = error;
        if (response && status === 400) {
          return toast.error(
            data.Error_Log.userName
            || data.Error_Log.userEmail
            || data.Error_Log.userPassword
          );
        }
        if (response && status === 401) {
          return toast.error('Incorrect login credentials');
        }
        if (response && status === 404) {
          return toast.error('Incorrect login credentials');
        }
        if (response && status === 406) {
          return toast.error(data.status);
        }
        toast.error('Unknown error');
      });
    this.setState({
      username: '',
      email: '',
      password: ''
    });
  };

  render() {
    const {
      username,
      email,
      password,
      usernameErrorMessage,
      passwordErrorMessage,
      emailErrorMessage,
    } = this.state;

    const buttonDisabled = this.hasFormErrors() || this.hasEmptyRequiredFields();

    const { tooglePassword, showPassword, toggleLoginSignup } = this.props;
    return (
      <div
        id="register"
        className="animate form"
      >
        <form
          className="
        login zoom"
          id="signupForm"
          autoComplete="on"
        >
          <div className="img_login">
            <h1> Sign up </h1>
          </div>

          <div className="loginParticulars">
            <div><b>Username</b></div>
            <small className="password-error-message">
            &nbsp;
              {usernameErrorMessage}
            </small>
            <input
              type="text"
              id="username"
              className={usernameErrorMessage && 'red-bg'}
              placeholder="Enter Username"
              name="username"
              value={username}
              onBlur={this.onUsernameBlur}
              onChange={this.handleOnChangeUsername}
            />
            <div id="emaildiv"><b>Email Address</b></div>
            <small className="password-error-message">
            &nbsp;
              {emailErrorMessage}
            </small>
            <input
              type="text"
              id="email"
              className={`email ${emailErrorMessage && 'red-bg'}`}
              placeholder="Enter Email Address"
              name="Email"
              value={email}
              onBlur={this.onEmailBlur}
              onChange={this.handleOnChangeEmail}
            />
            <div><b>Password</b></div>
            <small className="password-error-message">
            &nbsp;
              {passwordErrorMessage}
            </small>
            <input
              type={showPassword ? 'text' : 'password'}
              className={`password ${passwordErrorMessage && 'red-bg'}`}
              id="password"
              placeholder="Enter Password"
              name="password"
              value={password}
              onBlur={this.onPasswordBlur}
              onChange={this.handleOnChangePassword}
            />
          </div>
          <div>
            <input
              type="checkbox"
              checked={showPassword}
              className="showPassword"
              id="showPassword"
              name="remember"
              onChange={tooglePassword}
            />
        Show Password
          </div>

          <Button
            className="signupButton"
            id="submitBtn"
            type="submit"
            disabled={buttonDisabled}
            onClick={this.handleOnClickSignup}
            value="Sign up"
          />


          <div className="container">

            <div className="change_link">
                    Already a member ?
              <a
                href="#tologin"
                className="to_register"
                id="to_login"
                onClick={toggleLoginSignup}
              >
                    Go and log in
              </a>
            </div>
          </div>
        </form>

      </div>

    );
  }
}
Signup.propTypes = {
  toggleLoginSignup: PropTypes.func,
  tooglePassword: PropTypes.func,
  showPassword: PropTypes.bool,
  signup: PropTypes.func,
  history: PropTypes.object
};
