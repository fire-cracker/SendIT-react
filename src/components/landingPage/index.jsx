import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { ToastContainer } from 'react-toastify';

import LoginForm from './LoginForm';
import SigninForm from './SignupForm';
import { login } from '../../redux/actions/login';
import { signup } from '../../redux/actions/signup';
import './style.css';

export class LandingPage extends Component {
  state = {
    showLogin: true,
    showPassword: false
  };

  toggleLoginSignup = () => {
    this.setState(prevState => ({
      showLogin: !prevState.showLogin
    }));
  };

  tooglePassword = () => {
    this.setState(prevState => ({
      showPassword: !prevState.showPassword
    }));
  }

  render() {
    const { login, signup, history } = this.props;
    const { showLogin, showPassword } = this.state;
    return (
      <div className="whole">
        <ToastContainer />
        <div className="heroWhite" id="heroWhite">
          <h1 className="hero"> A NEW WAY TO SEND</h1>
          <p className="heroText">
            We make millions of deliveries every day.
            But we deliver more than just packages.
            We deliver trust and peace of mind. You can SendIt and count on us!
          </p>
        </div>
        {/* <!-- User Login/SignUp Section --> */}
        <div className="containerDemo">
          <div id="wrapper">
            {showLogin
              ? (
                <LoginForm
                  toggleLoginSignup={this.toggleLoginSignup}
                  login={login}
                  history={history}
                  showPassword={showPassword}
                  tooglePassword={this.tooglePassword}
                />
              )
              : (
                <SigninForm
                  toggleLoginSignup={this.toggleLoginSignup}
                  signup={signup}
                  history={history}
                  showPassword={showPassword}
                  tooglePassword={this.tooglePassword}
                />
              )}

          </div>
        </div>
      </div>
    );
  }
}
LandingPage.propTypes = {
  login: PropTypes.func,
  signup: PropTypes.func,
  history: PropTypes.object
};
const mapStateToProps = ({ login, signup }) => ({ login, signup });

const mapDispatchToProps = ({
  login,
  signup
});

export default connect(mapStateToProps, mapDispatchToProps)(LandingPage);
