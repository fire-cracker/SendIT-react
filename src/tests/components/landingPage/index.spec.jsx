import React from 'react';
import { mount, shallow } from 'enzyme';
import { toast } from 'react-toastify';
import axios from '../../../utils/axiosConfig';

import { LandingPage } from '../../../components/landingPage';
import Login from '../../../components/landingPage/LoginForm';
import Signup from '../../../components/landingPage/SignupForm';

jest.mock('../../../utils/axiosConfig');
jest.mock('react-toastify');
jest.mock('../../../redux/actions/login');
jest.mock('../../../redux/actions/signup');

let wrapper;
const preventDefault = jest.fn();

beforeEach(() => {
  wrapper = shallow(<LandingPage />);
});

describe('Tests for the Landing Page', () => {
  it('Should render the landing page', () => {
    wrapper = mount(<LandingPage />);
    expect(wrapper.length).toBe(1);
    expect(wrapper).toMatchSnapshot();
    wrapper.find('a').at(0).simulate('click');
  });
});
describe('Tests for the login Page', () => {
  it('Should render the login page', () => {
    wrapper = shallow(<Login />);
    expect(wrapper.length).toBe(1);
    expect(wrapper).toMatchSnapshot();
  });

  it('should set state onChange', () => {
    wrapper = mount(<Login />);
    wrapper.find('input').at(0).simulate('change', {
      target: { value: 'm' }
    });
    expect(wrapper.state('email')).toBe('m');

    wrapper.find('input').at(1).simulate('change', {
      target: { value: 'm' }
    });
    expect(wrapper.state('password')).toBe('m');
  });

  it('should test for onBlur', () => {
    wrapper = mount(<Login />);
    wrapper.find('input').at(0).simulate('change', {
      target: { value: '' }
    });
    wrapper.find('input').at(0).simulate('blur');
    expect(wrapper.state('emailErrorMessage')).toBe('Please enter your email.');

    wrapper.find('input').at(0).simulate('change', {
      target: { value: 'oyedeji' }
    });
    wrapper.find('input').at(0).simulate('blur');
    expect(wrapper.state('emailErrorMessage')).toBe('email is invalid.');

    wrapper.find('input').at(1).simulate('change', {
      target: { value: '' }
    });
    wrapper.find('input').at(1).simulate('blur');
    expect(wrapper.state('passwordErrorMessage')).toBe(
      'Please enter your password.'
    );

    wrapper.find('input').at(1).simulate('change', {
      target: { value: 'ghg ty' }
    });
    wrapper.find('input').at(1).simulate('blur');
    expect(wrapper.state('passwordErrorMessage')).toBe(
      'Use numbers and letters for password.'
    );

    wrapper.find('input').at(1).simulate('change', {
      target: { value: 'oyede' && 'oyedejipeaceoyededddndkd' }
    });
    wrapper.find('input').at(1).simulate('blur');
    expect(wrapper.state('passwordErrorMessage')).toBe(
      'Password must between 8 and 15 characters.'
    );
  });

  it('should login user if user provides correct login credentials',
    async () => {
      const props = {
        login: jest.fn().mockImplementation(() => Promise.resolve({}))
      };
      wrapper = shallow(<Login {...props} />);

      wrapper.find('input').at(0).simulate('change', {
        target: { value: 'oyedejipeace@yahoo.com' }
      });

      wrapper.find('input').at(1).simulate('change', {
        target: { value: 'oyedejipeace' }
      });

      wrapper.find('Button').simulate('click', { preventDefault });
      expect(props.login).toHaveBeenCalled();
    });

  it('should throw an error user login credentials are incorrect', async () => {
    const props = {
      login: jest.fn().mockImplementation(() => Promise.reject({}))
    };
    wrapper = mount(<Login {...props} />);

    wrapper.find('Button').simulate('click', { preventDefault });
    axios.post.mockResolvedValue(props.login);
  });

  it('should throw toast error if error code is 400', async () => {
    toast.error = jest.fn();
    const props = {
      login: jest.fn().mockImplementation(() => Promise.reject({
        response: {
          status: 400
        }
      }))
    };
    wrapper = shallow(<Login {...props} />);

    wrapper.find('Button').simulate('click', { preventDefault });
    axios.post.mockResolvedValue(props.login);
  });

  it('should throw toast error if error code is 401', async () => {
    const props = {
      login: jest.fn().mockImplementation(() => Promise.reject({
        response: {
          status: 401
        }
      }))
    };
    wrapper = shallow(<Login {...props} />);

    wrapper.find('Button').simulate('click', { preventDefault });
    axios.post.mockResolvedValue(props.login);
  });

  it('should throw toast error if error code is 404', async () => {
    const props = {
      login: jest.fn().mockImplementation(() => Promise.reject({
        response: {
          status: 404
        }
      }))
    };
    wrapper = shallow(<Login {...props} />);

    wrapper.find('Button').simulate('click', { preventDefault });
    axios.post.mockResolvedValue(props.login);
  });

  it('should throw toast error for all other errors', async () => {
    const props = {
      login: jest.fn().mockImplementation(() => Promise.reject({}))
    };
    wrapper = shallow(<Login {...props} />);

    wrapper.find('Button').simulate('click', { preventDefault });
    axios.post.mockResolvedValue(props.login);
  });
});

describe('Tests for the signup Page', () => {
  it('Should render the signup page', () => {
    wrapper = shallow(<Signup />);
    expect(wrapper.length).toBe(1);
    expect(wrapper).toMatchSnapshot();
  });

  it('should set state onChange', () => {
    wrapper = mount(<Signup />);

    wrapper.find('input').at(0).simulate('change', {
      target: { value: 'm' }
    });
    expect(wrapper.state('username')).toBe('m');

    wrapper.find('input').at(1).simulate('change', {
      target: { value: 'm' }
    });
    expect(wrapper.state('email')).toBe('m');

    wrapper.find('input').at(2).simulate('change', {
      target: { value: 'm' }
    });
    expect(wrapper.state('password')).toBe('m');
  });

  it('should test for onBlur', () => {
    wrapper = mount(<Signup />);

    wrapper.find('input').at(1).simulate('change', {
      target: { value: '' }
    });
    wrapper.find('input').at(0).simulate('blur');
    expect(wrapper.state('usernameErrorMessage')).toBe('Please enter your username.');

    wrapper.find('input').at(0).simulate('change', {
      target: { value: 'oyedeji gh' }
    });
    wrapper.find('input').at(0).simulate('blur');
    expect(wrapper.state('usernameErrorMessage')).toBe('Use numbers and letters for password.');

    wrapper.find('input').at(0).simulate('change', {
      target: { value: 'nj' && 'oyedejipeaceoyededddndkd' }
    });
    wrapper.find('input').at(0).simulate('blur');
    expect(wrapper.state('usernameErrorMessage')).toBe('Password must between 8 and 15 characters.');

    wrapper.find('input').at(1).simulate('change', {
      target: { value: '' }
    });
    wrapper.find('input').at(1).simulate('blur');
    expect(wrapper.state('emailErrorMessage')).toBe('Please enter your email.');

    wrapper.find('input').at(1).simulate('change', {
      target: { value: 'oyedeji' }
    });
    wrapper.find('input').at(1).simulate('blur');
    expect(wrapper.state('emailErrorMessage')).toBe('email is invalid.');

    wrapper.find('input').at(2).simulate('change', {
      target: { value: '' }
    });
    wrapper.find('input').at(2).simulate('blur');
    expect(wrapper.state('passwordErrorMessage')).toBe(
      'Please enter your password.'
    );

    wrapper.find('input').at(2).simulate('change', {
      target: { value: 'ghg ty' }
    });
    wrapper.find('input').at(2).simulate('blur');
    expect(wrapper.state('passwordErrorMessage')).toBe(
      'Use numbers and letters for password.'
    );

    wrapper.find('input').at(2).simulate('change', {
      target: { value: 'oyede' && 'oyedejipeaceoyededddndkd' }
    });
    wrapper.find('input').at(2).simulate('blur');
    expect(wrapper.state('passwordErrorMessage')).toBe(
      'Password must between 8 and 15 characters.'
    );
  });

  it('should signup user if user provides correct signup credentials',
    async () => {
      const props = {
        signup: jest.fn().mockImplementation(() => Promise.resolve({}))
      };
      wrapper = shallow(<Signup {...props} />);

      wrapper.find('input').at(0).simulate('change', {
        target: { value: 'oyedejipeace@yahoo.com' }
      });

      wrapper.find('input').at(1).simulate('change', {
        target: { value: 'oyedejipeace' }
      });

      wrapper.find('Button').simulate('click', { preventDefault });
      expect(props.signup).toHaveBeenCalled();
    });

  it('should throw an error user signup credentials are incorrect', async () => {
    const props = {
      signup: jest.fn().mockImplementation(() => Promise.reject({}))
    };
    wrapper = shallow(<Signup {...props} />);

    wrapper.find('Button').simulate('click', { preventDefault });
    axios.post.mockResolvedValue(props.signup);
  });

  it('should throw toast error if error code is 400', async () => {
    const props = {
      signup: jest.fn().mockImplementation(() => Promise.reject({
        response: {
          status: 400
        }
      }))
    };
    wrapper = shallow(<Signup {...props} />);

    wrapper.find('Button').simulate('click', { preventDefault });
    axios.post.mockResolvedValue(props.signup);
  });

  it('should throw toast error if error code is 401', async () => {
    const props = {
      signup: jest.fn().mockImplementation(() => Promise.reject({
        response: {
          status: 401
        }
      }))
    };
    wrapper = shallow(<Signup {...props} />);

    wrapper.find('Button').simulate('click', { preventDefault });
    axios.post.mockResolvedValue(props.signup);
  });

  it('should throw toast error if error code is 404', async () => {
    const props = {
      signup: jest.fn().mockImplementation(() => Promise.reject({
        response: {
          status: 404
        }
      }))
    };
    wrapper = shallow(<Signup {...props} />);

    wrapper.find('Button').simulate('click', { preventDefault });
    axios.post.mockResolvedValue(props.signup);
  });
  it('should throw toast error if error code is 406', async () => {
    const props = {
      signup: jest.fn().mockImplementation(() => Promise.reject({
        response: {
          status: 406
        }
      }))
    };
    wrapper = shallow(<Signup {...props} />);

    wrapper.find('Button').simulate('click', { preventDefault });
    axios.post.mockResolvedValue(props.signup);
  });
  it('should throw toast error for all other errors', async () => {
    const props = {
      signup: jest.fn().mockImplementation(() => Promise.reject({}))
    };
    wrapper = shallow(<Signup {...props} />);

    wrapper.find('Button').simulate('click', { preventDefault });
    axios.post.mockResolvedValue(props.signup);
  });
});
