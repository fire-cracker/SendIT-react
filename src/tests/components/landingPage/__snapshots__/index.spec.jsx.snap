// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for the Landing Page Should render the landing page 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`Tests for the login Page Should render the login page 1`] = `
<div
  className="animate form"
  id="login"
>
  <form
    autoComplete="on"
    className="login zoom"
    id="LoginForm"
  >
    <div
      className="img_login"
    >
      <h1>
        Log in
      </h1>
    </div>
    <div
      className="loginParticulars"
    >
      <div>
        <b>
          Email Address
        </b>
      </div>
      <small
        className="password-error-message"
      >
         
      </small>
      <input
        className="email "
        id="loginEmail"
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Enter Email Address"
        type="text"
        value=""
      />
      <div>
        <b>
          Password
        </b>
      </div>
      <small
        className="password-error-message"
      >
         
      </small>
      <input
        className="password "
        id="loginPassword"
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Enter Password"
        type="password"
        value=""
      />
      <div>
        <input
          className="showPassword"
          id="loginShowPassword"
          name="remember"
          type="checkbox"
        />
        Show Password
      </div>
      <Button
        className="loginButton"
        disabled={true}
        id="loginBtn"
        onClick={[Function]}
        type="submit"
        value="Login"
      />
      <div
        className="container"
      >
        <div
          className="change_link"
        >
          Not a member yet ?
          <a
            className="to_register"
            href="#toregister"
            id="to_register"
          >
            Join us
          </a>
        </div>
      </div>
    </div>
  </form>
</div>
`;

exports[`Tests for the signup Page Should render the signup page 1`] = `
<div
  className="animate form"
  id="register"
>
  <form
    autoComplete="on"
    className=" login zoom"
    id="signupForm"
  >
    <div
      className="img_login"
    >
      <h1>
         Sign up 
      </h1>
    </div>
    <div
      className="loginParticulars"
    >
      <div>
        <b>
          Username
        </b>
      </div>
      <small
        className="password-error-message"
      >
         
      </small>
      <input
        className=""
        id="username"
        name="username"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Enter Username"
        type="text"
        value=""
      />
      <div
        id="emaildiv"
      >
        <b>
          Email Address
        </b>
      </div>
      <small
        className="password-error-message"
      >
         
      </small>
      <input
        className="email "
        id="email"
        name="Email"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Enter Email Address"
        type="text"
        value=""
      />
      <div>
        <b>
          Password
        </b>
      </div>
      <small
        className="password-error-message"
      >
         
      </small>
      <input
        className="password "
        id="password"
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Enter Password"
        type="password"
        value=""
      />
    </div>
    <div>
      <input
        className="showPassword"
        id="showPassword"
        name="remember"
        type="checkbox"
      />
      Show Password
    </div>
    <Button
      className="signupButton"
      disabled={true}
      id="submitBtn"
      onClick={[Function]}
      type="submit"
      value="Sign up"
    />
    <div
      className="container"
    >
      <div
        className="change_link"
      >
        Already a member ?
        <a
          className="to_register"
          href="#tologin"
          id="to_login"
        >
          Go and log in
        </a>
      </div>
    </div>
  </form>
</div>
`;
